core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -199,6 +202,7 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         }
 
         TableHandle newTable;
+        Optional<TablePartitioning> newTablePartitioning;
```

Comment:
It is not obvious why this cannot be just boolean. Is it just for having na "unset" value for sake of optimizer rule pattern upcoming in one of followup commits
Add comment please.

Reply:
The thing is `PushPrediacteIntoTableScan` and `PushProjectionIntoTableScan` are executed after `AddExchanges` which plans according to partitioning. If the partitioning was to change, plan might be invalid.

Added comment

Reply:
Thanks :) 
Though I misplaced the review comment. I wanted to put it by `Optional<Boolean> useConnectorTablePartitioning` :) what about that place.

Reply:
It's optional so that it's rule responsibility to set it accordingly. Otherwise there would be some implicit default (independent of feature config). It's also more explicit that partitioning has been fixed for a table scan node.